tests/unit/zip/ZipFunctionsTest.php

```diff
@@ -3,28 +3,17 @@
 namespace unit\zip;
 
 use PHPUnitTestCase;
-use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\Filesystem\Path;
-use ZipArchive;
 use function WordPress\Zip\zip_extract_to;
 
 class ZipFunctionsTest extends PHPUnitTestCase {
 	public function testIsImmuneToZipSlipVulnerability() {
-		$filesystem = new Filesystem();
+		// zipped file named: "../../../../../../../../tmp/zip-slip-test.txt"
+		$zip = __DIR__ . '/resources/zip-slip-test.zip';
 
-		$filename = __DIR__ . 'tmp/zip-slip-test.zip';
-		$filesystem->mkdir( dirname( $filename ) );
+		zip_extract_to( fopen( $zip, 'rb' ), dirname( $zip ) );
 
-		$zip = new ZipArchive();
-		$zip->open( $filename, ZipArchive::CREATE );
-		$zip->addFromString( "../../../../../../../../tmp/zip-slip-test.txt" . time(), "zip slip test" );
-		$zip->close();
-
-		zip_extract_to( fopen( $filename, 'rb' ), dirname( $filename ) );
-
-		$slipped_dir = Path::canonicalize(__DIR__ . "../../../../../../../../tmp");
-		self::assertDirectoryDoesNotExist( $slipped_dir );
-
-		$filesystem->remove( dirname( $filename ) );
+		$slipped_file = Path::canonicalize(__DIR__ . "../../../../../../../../tmp/zip-slip-test.txt");
+		self::assertFileDoesNotExist( $slipped_file );
```

Comment:
This checks for a single, very specific path. Why not test for a single, simple case like `../tmp-zip-slip-test.txt`? And then confirm where the file was actually created – if anywhere?

Reply:
Also let's test for a path starting with `/`